crates/crypto/src/hash/mod.rs

```diff
@@ -39,12 +40,32 @@ impl FromStr for HashAlgorithm {
     }
 }
 
-pub trait SupportedDigest: Digest + private::Sealed {
+static EMPTY_TREE_HASH: Lazy<Vec<Hash<Sha256>>> = Lazy::new(|| {
+    let mut v: Vec<Hash<Sha256>> = Vec::with_capacity(257);
+    fn empty_tree_hash<D: SupportedDigest>(v: &mut Vec<Hash<D>>, height: u32) -> Hash<D> {
+        let hash: Hash<D> = if height == 0 {
+            Hash::of("")
+        } else {
+            let last_hash = empty_tree_hash(v, height - 1);
+            Hash::of(Hash::<Sha256>::of((0b1, &last_hash, &last_hash)))
+        };
+        v.push(hash.clone());
+        hash
+    }
+    empty_tree_hash(&mut v, 256);
+    v
+});
+
+pub trait SupportedDigest: Digest + private::Sealed + Sized + std::fmt::Debug {
```

Comment:
why is there a `std::fmt::Debug` bound here?
it shouldn't be necessary since `Hash<D> where D: SupportedDigest` impls `Debug`.